subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -157,10 +168,10 @@ int dfu_target_mcuboot_init(size_t file_size, int img_num, dfu_target_callback_t
 		return -EFBIG;
 	}
 
-	flash_dev = device_get_binding(secondary_dev_name[img_num]);
-	if (flash_dev == NULL) {
-		LOG_ERR("Failed to get device '%s'",
-			secondary_dev_name[img_num]);
+	flash_dev = secondary_dev[img_num];
+	if (!device_is_ready(flash_dev)) {
```

Comment:
I think we still need to null-check here (besides device_is_ready), because we use `DEVICE_DT_GET_OR_NULL` macro which can return null if someone adds `status = "disabled"` in the DTS.

Reply:
Nevermind, `device_is_ready` already does that.

Reply:
`device_is_ready` [checks](https://github.com/nrfconnect/sdk-zephyr/blob/482dcc7865f7d3a458f061a1d8276c5801eed1e4/kernel/device.c#L153-L159) if the parameter is `NULL`.